WebApiCoreSeed.UnitTests/Services/UserServiceTests.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Threading.Tasks;
+using WebApiCoreSeed.Data.EF;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.Domain.Services;
+using Xunit;
+
+namespace WebApiCoreSeed.UnitTests.Services
+{
+    public class UserServiceTests
+    {
+        [Fact]
+        public async Task GetByIdAsync_ShouldGetUserById_WhenUserIdExists()
+        {
+            // Arrange
+            var optionsBuilder = new DbContextOptionsBuilder<WebApiCoreSeedContext>();
+            optionsBuilder.UseInMemoryDatabase();
+            var createdUser = new User { Id = new Guid("d58893fd-3836-4308-b840-85f4fe548264"), FirstName = "John", LastName = "Doe", Email = "JohnDoe@makinsense.com", UserName = "JohnDoe" };
+
+            User retrievedUser = null;
+            using (var context = new WebApiCoreSeedContext(optionsBuilder.Options))
+            {
+                context.Add(createdUser);
+                context.SaveChanges();
+                var userService = new UserService(context);

```

Comment:
I am not sure if it is better or not for your test, but if you want a more realistic test you could use two different contexts:

```csharp
using (var context = new WebApiCoreSeedContext(optionsBuilder.Options))
{
    context.Add(createdUser);
    context.SaveChanges();
}
using (var context = new WebApiCoreSeedContext(optionsBuilder.Options))
{
    var userService = new UserService(context);
    // Act
    retrievedUser = await userService.GetByIdAsync(createdUser.Id);
}
```

In this way, `createdUser` and `retrievedUser` will be different objects with the same (or similar) data. Currently, in your test, both variables are pointing to the same object.